src/external.ml

```diff
@@ -114,3 +120,38 @@ let runExternalProgram cmd =
          else "\n\n" ^ Util.process_status_to_string returnValue)))
       (* Stop typechechecker from complaining about non-exhaustive pattern above *)
       | _ -> assert false)
+
+let runExternalProgramArgs cmd args =
```

Comment:
It's probably OK to just remove `runExternalProgram` completely and keep just `runExternalProgramArgs`.

The current `runExternalProgram` is used in very few locations and it may be worthwhile just changing those invocations to use the args-variant.

Reply:
I looked into this before, and I think it's tricky, because currently `runExternalProgram` is used for example for `merge`. Now, I have the following setting:

```
merge = emacsclient -q --eval '(ediff-merge-files-with-ancestor "CURRENT1" "CURRENT2" "OLD" nil "NEW")'
```

This won't work if it's parsed as a series of space-separated tokens; the parser needs to understand quoting. For `copyprog` I don't think this is a problem, as its syntax is fixed: the arguments will always be added to the end (and as we've said elsewhere, using anything other than rsync is dubious). For `merge` and `merge2`, they are substituted into the middle.

While it would be nice to avoid calling the shell, I don't think swapping the user's shell for a custom quote parser is a good idea. For this sort of case, it would be better IMO to change `merge` so that it too puts its arguments in a fixed place, and if you want something like my complex setting above you have to write a short shell-script or similar.

Reply:
You're right, we can't sensibly parse values of such preferences.

I just noticed that something like this may happen with the remote shell args as well:
https://github.com/bcpierce00/unison/blob/a66be8529f75abfeb66b8a54e72a7a1df3d527d1/src/remote.ml#L1817-L1832
(here, `create_process` and `exec*` functions are used)

The value of `shellCmdArgs` comes from a user preference and is later simply split by spaces. Nothing to do with this PR. I just noticed that the same issue lurks within that code, although mostly theoretical. As you've shown, for the merge preference this issue is very much real.

Let's keep these functions separate, as they are.